resources/pdns_recursor_config.rb

```diff
@@ -72,31 +72,34 @@
   directory new_resource.socket_dir do
     owner new_resource.run_user
     group new_resource.run_group
-    mode '0755'
-    recursive true
-    action :create
-  end
-
-  directory recursor_instance_dir do
-    owner 'root'
-    group 'root'
-    mode '0755'
+    # When using service_manager the 'socket-dir' has to be writable for the 'set-gid'
+    # in order to start the service:
+    # Issue: https://github.com/PowerDNS/pdns/issues/4826
+    mode '0775'
     recursive true
     action :create
   end
 
-  template "#{recursor_instance_dir}/recursor.conf" do
+  template "#{new_resource.config_dir}/recursor-#{new_resource.instance_name}.conf" do
     source new_resource.source
     cookbook new_resource.cookbook
     owner 'root'
     group 'root'
     mode '0640'
     variables(
-      config_dir: recursor_instance_dir,
       socket_dir: new_resource.socket_dir,
       setuid: new_resource.setuid,
       setgid: new_resource.setgid,
       variables: new_resource.variables
       )
   end
 end
+
+action_class.class_eval do
+  include ::PdnsResource::Helpers
+  include ::PdnsRecursorResource::Helpers
+
+  def whyrun_supported?
```

Comment:
This should have already been defined for us by Chef. The 12.5+ syntax includes this method.